src/components/common/Calculator.js

```diff
@@ -45,6 +48,36 @@ const radToDegree = (rad) => {
   return rad * (180 / Math.PI);
 };
 
+// use special cases because of rounding error
+const getSpecialCase = (angle, fn, isRadian) => {
+  let quarter;
+  let remaining;
+  if (isRadian) {
+    // radian
+    quarter = Math.ceil((angle / (Math.PI / 4)) * 45) % 360;
+    remaining = angle % (Math.PI / 4);
+  } else {
+    // degree
+    quarter = angle % 360;
+    remaining = angle % 45;
+  }
+
+  if (angle < 0) {
```

Comment:
Why are you doing this?

Reply:
ex: `cos(-90)` Since previous computations would be negative and won't map to my special cases mapping, I need to make it positive.